bin/lock_dev_tool.ml

```diff
@@ -185,34 +185,51 @@ let lockdir_status dev_tool =
     (match Lock_dir.read_disk dev_tool_lock_dir with
      | Error _ -> Memo.return `No_lockdir
      | Ok { packages; _ } ->
-       (match Dune_pkg.Dev_tool.needs_to_build_with_same_compiler_as_project dev_tool with
-        | false -> Memo.return `Lockdir_ok
-        | true ->
-          let* platform =
-            Pkg.Pkg_common.poll_solver_env_from_current_system ()
-            |> Memo.of_reproducible_fiber
-          in
-          let packages = Lock_dir.Packages.pkgs_on_platform_by_name packages ~platform in
-          (match Package_name.Map.find packages compiler_package_name with
-           | None -> Memo.return `No_compiler_lockfile_in_lockdir
-           | Some { info; _ } ->
-             let+ ocaml_compiler_version = locked_ocaml_compiler_version () in
-             (match Package_version.equal info.version ocaml_compiler_version with
-              | true -> `Lockdir_ok
-              | false ->
-                `Dev_tool_needs_to_be_relocked_because_project_compiler_version_changed
-                  (User_message.make
-                     [ Pp.textf
-                         "The version of the compiler package (%S) in this project's \
-                          lockdir has changed to %s (formerly the compiler version was \
-                          %s). The dev-tool %S will be re-locked and rebuilt with this \
-                          version of the compiler."
-                         (Package_name.to_string compiler_package_name)
-                         (Package_version.to_string ocaml_compiler_version)
-                         (Package_version.to_string info.version)
-                         (Dune_pkg.Dev_tool.package_name dev_tool
-                          |> Package_name.to_string)
-                     ])))))
+       let* platform =
+         Pkg.Pkg_common.poll_solver_env_from_current_system ()
+         |> Memo.of_reproducible_fiber
+       in
+       let packages = Lock_dir.Packages.pkgs_on_platform_by_name packages ~platform in
+       let package_name = Dune_pkg.Dev_tool.package_name dev_tool in
+       (match Package_name.Map.find packages package_name with
+        | None ->
+          Memo.return
+            (`Lockdir_missing_entry_for_tool
+                (User_message.make
+                   [ Pp.textf
+                       "The lock directory for the tool %S exists but does not contain a \
+                        lockfile for the package %S. This may indicate that the lock \
+                        directory has been tampered with. Please avoid making manual \
+                        changes to tool lock directories. The tool will now be relocked."
```

Comment:
Should this be a warning? It sounds like a warning...

Reply:
Good point. Changed to a warning.